src/v/cluster/topics_frontend.cc

```diff
@@ -62,6 +62,26 @@
 #include <sstream>
 #include <system_error>
 
+namespace {
+
+std::optional<std::string_view> get_enterprise_feature(
+  const cluster::custom_assignable_topic_configuration& cfg) {
+    if (
+      cfg.cfg.properties.shadow_indexing
+      != model::shadow_indexing_mode::disabled) {
+        return "tiered storage";
+    }
```

Comment:
I can't remember if we've already discussed this, but do we want to check for `redpanda.remote.delete` here as well? It doesn't make sense to add remote delete to a topic without remote read/write, but it is on the list of proposed sanctions.

Reply:
> I can't remember if we've already discussed this, but do we want to check for redpanda.remote.delete here as well

The problem is that `redpanda.remote.delete` is always on by default, checking it here causes more problems than it solves.